server/chat-plugins/datasearch.js

```diff
@@ -1091,36 +1080,53 @@ function runMovesearch(target, cmd, canAll, message) {
 		return {reply: "No search parameters other than 'all' were found. Try '/help movesearch' for more information on this command."};
 	}
 
-	let dex = {};
-	if (targetMons.length) {
-		let intersectionOfLearnsets = [];
-		for (let mon in lsetData) {
-			if (intersectionOfLearnsets.length) {
-				intersectionOfLearnsets = intersectionOfLearnsets.filter(move => {
-					return lsetData[mon][move] !== undefined;
-				});
-			} else {
-				intersectionOfLearnsets = Object.keys(lsetData[mon]);
-			}
-		}
+	const dex = {};
+	for (let move in Dex.data.Movedex) {
+		dex[move] = Dex.getMove(move);
+	}
+	delete dex.magikarpsrevenge;
+
+	const getFullLearnsetOfPokemon = (template) => {
+		if (!template.learnset) template = Dex.getTemplate(template.baseSpecies);
+		let lsetData = new Set(Object.keys(template.learnset));
 
-		let finalizedLearnset = {};
-		for (let move of intersectionOfLearnsets) {
-			finalizedLearnset[move] = Object.values(lsetData)[0][move];
+		while (template.prevo) {
+			template = Dex.getTemplate(template.prevo);
+			lsetData = new Set([...lsetData, ...Object.keys(template.learnset)]);
 		}
 
-		for (let move in finalizedLearnset) {
-			dex[move] = Dex.getMove(move);
+		return lsetData;
+	};
+
+	// Since we assume we have no target mons at first
+	// then the valid moveset we can search is the entire dex.
```

Comment:
'entire dex' confused me: entire movedex / set of all moves / etc would read better I think?

Reply:
Changed to "set of all moves".